administrator/components/com_menus/models/fields/modal/menu.php

```diff
@@ -0,0 +1,252 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_menus
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('JPATH_BASE') or die;
+
+/**
+ * Supports a modal menu item picker.
+ *
+ * @since  3.7
+ */
+class JFormFieldModal_Menu extends JFormField
+{
+	/**
+	 * The form field type.
+	 *
+	 * @var     string
+	 * @since   3.7
+	 */
+	protected $type = 'Modal_Menu';
+
+	/**
+	 * Method to get the field input markup.
+	 *
+	 * @return  string  The field input markup.
+	 *
+	 * @since   3.7
+	 */
+	protected function getInput()
+	{
+		$allowEdit  = ((string) $this->element['edit'] == 'true') ? true : false;
+		$allowClear = ((string) $this->element['clear'] != 'false') ? true : false;
+
+		// Load language
+		JFactory::getLanguage()->load('com_menus', JPATH_ADMINISTRATOR);
+
+		// The active article id field.
+		$value = (int) $this->value > 0 ? (int) $this->value : '';
+
+		// Build the script.
+		$script = array();
+
+		// Select button script
+		$script[] = '	function jSelectMenu_' . $this->id . '(id, title, object) {';
```

Comment:
You could really save $this->id into a local js variable, having that 20 times inline with php looks not so nice imo :-)

`var id = ' . $this->id '; `